lib/node_modules/@stdlib/utils/parse-ndjson/test/test.js

```diff
@@ -0,0 +1,178 @@
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var noop = require( '@stdlib/utils/noop' );
+var parseNDJSON = require( './../lib' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof parseNDJSON, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function throws if not provided a string', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		3.14,
+		NaN,
+		true,
+		null,
+		void 0,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws a type error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			parseNDJSON( value );
+		};
+	}
+});
+
+tape( 'the function throws if not provided a string (reviver)', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		3.14,
+		NaN,
+		true,
+		null,
+		void 0,
+		[],
+		{},
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws a type error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			parseNDJSON( value, noop );
+		};
+	}
+});
+
+tape( 'the function throws if provided a reviver argument which is not a function', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'5',
+		3.14,
+		NaN,
+		true,
+		null,
+		void 0,
+		[],
+		{}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws a type error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			parseNDJSON( '{"a":"b"}\n{"c": "d"}', value );
+		};
+	}
+});
+
+tape( 'the function returns an array of parsed values if provided valid NDJSON', function test( t ) {
+	var expected;
+	var actual;
+
+	expected = [
+		{
+			'beep': 'boop'
+		},
+		{
+			'example': 42
+		},
+		{
+			'data': [ 1, 2, 3 ]
+		}
+	];
+	actual = parseNDJSON( '{"beep":"boop"}\n{"example":42}\n{"data":[1,2,3]}' );
+	t.deepEqual( actual, expected, 'deep equal' );
+
+	expected = [];
+	actual = parseNDJSON('');
+
+	t.deepEqual( actual, expected, 'returns an empty array' );
+
+	t.end();
+});
+
+tape( 'the function returns an error if provided invalid NDJSON', function test( t ) {
+	var out = parseNDJSON('{"beep":"boop"');
+	t.equal( out instanceof SyntaxError, true, 'returns an error' );
+	t.end();
+});
+
+tape( 'the function supports providing a custom reviver function', function test( t ) {
+	var expected;
+	var actual;
+	var str;
+
+	function reviver( key, value ) {
+		if ( key === '' || key === 'beep' ) {
+			return ( typeof value === 'string' ) ? value.toUpperCase() : value;
+		}
+		return ( typeof value === 'number' ) ? value * 2 : value;
+	}
+
+	str = '{"beep":"boop"}\n{"value": 20}\n{"numbers": [1,2,3]}';
```

Comment:
We also need to account for a trailing newline. E.g., I should be able to read a file containing ndjson, followed by a trailing newline, and then pipe into this function and have it behave as expected.

Reply:
Got it, I'll implement it

Reply:
Implemented.